app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/NoCollectionsMessageViewHolder.kt

```diff
@@ -5,59 +5,210 @@
+            store,
+            interactor::onAddTabsToCollectionTapped,
+            interactor::onRemoveCollectionsPlaceholder
+        )
+    }
+}
+
+/**
+ * [NoCollectionsMessage] for displaying a message detailing the collections feature and
+ * allowing users to easily start creating their first.
+ *
+ * @param store to receive live updates to normal tabs.
+ * @param onAddTabsToCollectionTapped to create a new first collection.
+ * @param onRemoveCollectionsPlaceholder to remove empty collections message.
+ */
+
+@Composable
+fun NoCollectionsMessage(
+    store: BrowserStore,
+    onAddTabsToCollectionTapped: () -> Unit,
+    onRemoveCollectionsPlaceholder: () -> Unit,
+) {
+    val normalTabsState = store.observeAsComposableState { state -> state.normalTabs }
+
+    Box(
+        modifier = Modifier
+            .semantics(mergeDescendants = true) {}
+            .dashedBorder(
+                color = FirefoxTheme.colors.borderPrimary,
+                cornerRadius = 8.dp,
+                dashHeight = 2.dp,
+                dashWidth = 4.dp
             )
-            setOnClickListener {
-                interactor.onRemoveCollectionsPlaceholder()
-            }
-        }
+            .padding(
+                start = 16.dp,
+                bottom = 16.dp
+            )
+    ) {
+        Column(
+            Modifier
+                .padding(all = 16.dp)
+                .fillMaxWidth()
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                SectionHeader(
+                    text = stringResource(R.string.collections_header),
+                    modifier = Modifier
+                        .weight(1f)
+                        .wrapContentHeight(align = Alignment.Top)
+                )
 
-        store.flowScoped(viewLifecycleOwner) { flow ->
-            flow.map { state -> state.normalTabs.size }
-                .ifChanged()
-                .collect { tabs ->
-                    binding.addTabsToCollectionsButton.isVisible = tabs > 0
+                IconButton(
+                    onClick = { onRemoveCollectionsPlaceholder() }
+                ) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_close),
+                        contentDescription = stringResource(
+                            id = R.string.remove_home_collection_placeholder_content_description
+                        ),
+                        modifier = Modifier.size(24.dp),
+                        tint = FirefoxTheme.colors.textPrimary
+                    )
                 }
+            }
+            PrimaryText(
+                text = stringResource(R.string.no_collections_description2),
+                modifier = Modifier.fillMaxWidth(),
+                fontSize = 14.sp
+            )
```

Comment:
nit: we normally have an empty line between logic blocks and composable blocks (152 -> 153 and 157 -> 158)